elastic/apm-server
# Introduce model.Processor

## Motivation/summary

Introduce type `model.Processor` and field `model.APMEvent.Processor`, which controls the `processor.name` and `processor.event` fields. These fields are used for identifying the event type, used for routing to the appropriate index/data stream and also for monitoring purposes.

Processed event metrics have been moved to a dedicated `model.BatchProcessor`, and out of the model package; they use the Processor field to form the metric name.

Note: Several unused metrics have been removed: stack trace and stack frame counts.

## Checklist

~- [ ] Documentation has been updated~

## How to test these changes

Enable monitoring, ingest some events, check that `apm-server.processor.<event-type>.transformations` metrics are recorded.

## Related issues

elastic/apm-server#4410
elastic/apm-server#4120
elastic/apm-server#4715
